src/stores/index.js

```diff
@@ -358,6 +380,44 @@ const Store = Reflux.createStore({
     }
   },
 
+  /**
+   * Copies a favorite connection.
+   *
+   * @param {Connection} connection - The connection to select.
+   */
+  onCopyConnectionClicked(connection) {
+    const newConnection = new Connection();
+
+    newConnection.set(omit(connection.getAttributes({ props: true }), ['_id', 'color']));
+    newConnection.set({ name: `${connection.name} (copy)` });
+
+    this._addConnection(newConnection);
+  },
+
+
+  /**
+   * Discards favorite changes.
+   */
+  onFavoriteChangeDiscarded() {
+    this.state.connections.fetch({ success: () => this.trigger(this.state) });
```

Comment:
what happens on error here?

Reply:
We don't update the state the same as we did before to initialize the state. Take a look at `init()` method at the beginning of the file. This is a callback of the ampersand js if you want to add some custom logic after fetching documents.